go-controller/pkg/ovn/controller/services/utils.go

```diff
@@ -160,22 +167,41 @@ func createPerNodeVIPs(svcIPs []string, protocol v1.Protocol, sourcePort int32,
 	return nil
 }
 
+func createNodePortVIPs(service *v1.Service, isIpv6 bool, nodePortProtocol v1.Protocol, nodePort int32, eps util.LbEndpoints) error {
+	isPolicyLocal := util.ServiceExternalTrafficPolicyLocal(service)
+	if err := createPerNodePhysicalVIPs(isIpv6, nodePortProtocol, nodePort, eps, eps.Port, isPolicyLocal); err != nil {
+		return err
+	}
+	return nil
+}
+
 // createPerNodePhysicalVIPs adds load balancers on a per node basis for GR and worker switch LBs using physical IPs
-func createPerNodePhysicalVIPs(isIPv6 bool, protocol v1.Protocol, sourcePort int32, targetIPs []string, targetPort int32) error {
-	klog.V(5).Infof("Creating Node VIPs - %s, %d, [%v], %d", protocol, sourcePort, targetIPs, targetPort)
+func createPerNodePhysicalVIPs(isIPv6 bool, protocol v1.Protocol, sourcePort int32, eps util.LbEndpoints, targetPort int32, nodeLocal bool) error {
+	targetIPs := eps.IPs()
+	klog.V(5).Infof("Creating Node VIPs - %s, %d, [%v], %d %v", protocol, sourcePort, eps.IPs(), targetPort, nodeLocal)
 	// Each gateway has a separate load-balancer for N/S traffic
 	gatewayRouters, _, err := gateway.GetOvnGateways()
 	if err != nil {
 		return err
 	}
 
 	for _, gatewayRouter := range gatewayRouters {
+		// Get normal per GWR load-balancers
 		gatewayLB, err := gateway.GetGatewayLoadBalancer(gatewayRouter, protocol)
 		if err != nil {
 			klog.Errorf("Gateway router %s does not have load balancer (%v)",
 				gatewayRouter, err)
 			continue
 		}
+		// Get node-local per GWR load balancers only
+		if nodeLocal {
+			gatewayLB, err = gateway.GetGatewayLoadBalancer(fmt.Sprintf("%s_local", gatewayRouter), protocol)
```

Comment:
you overwrite gatewayLB here if nodeLocal, so there is no reason to execute line 190. You should move that to an else clause.